### PR TITLE
Heater demo example using Maven/BND.

### PR DESCRIPTION
**Pull request by dwoodard1**
@ctron @cdealti I had several people ask me at a recent Eclipse event about pure Maven and BND with Eclipse Kura. I know @ctron has written about this as well. Some are under the impression you must use Eclipse PDE/Tycho tooling in order to write bundles for Eclipse. 

As such, I ported the heater demo to use annotations and the maven BND plugin. I think it would be good to have this example the Kura repo for reference.

### CODE REVIEW

File: kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+ */
+@Designate(ocd = HeaterConfig.class)
+@Component(
+        immediate = true,
+        configurationPolicy = ConfigurationPolicy.REQUIRE,
+        property = "service.pid=org.eclipse.kura.demo.bnd.heater.Heater"
+)
+public class Heater implements ConfigurableComponent, CloudClientListener {
+
+    private static final Logger s_logger = LoggerFactory.getLogger(Heater.class);
```
Comment (ctron) [inline kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java]: Could we use the new style, simply use `logger`, also see other members.

Comment (dwoodard1) [reply kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java]: Yep, will make a commit shortly. I was using IntelliJ to make sure I was avoiding any Eclipse specific tooling. May be nice to include an IntelliJ code formatter that mimics what we have for Eclipse.

File: kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/pom.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        </plugins>
+
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.6.1</version>
+                    <configuration>
+                        <source>1.7</source>
```
Comment (markoer) [inline kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/pom.xml]: Maybe go for 1.8, since Kura now uses this version.

**Review by markoer — APPROVED** [review]

File: kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    }
+
+    @Modified
+    public void updated(Map<String, Object> properties) {
+        logger.info("Updated Heater...");
+
+        // store the properties received
+        this.properties = properties;
+        for (String s : properties.keySet()) {
+            logger.info("Update - " + s + ": " + properties.get(s));
```
Comment (ctron) [inline kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java]: I just spotted this one... Should better be `Update - {}: {}` ... like a few lines above.

**Review by ctron — APPROVED** [review]

**Comment by MMaiero** [comment]
@dwoodard1 Why this PR is failing the IP validation?

**Comment by dwoodard1** [comment]
@MMaiero I didn't sign the commit.

File: kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java [hunk 4]
```diff
@@ -299,7 +300,7 @@ private void doPublish() {
             this.cloudClient.publish(topic, payload, qos, retain);
             logger.info("Published to {} message: {}", topic, payload);
         } catch (Exception e) {
-            logger.error("Cannot publish topic: " + topic, e);
+            logger.error("Cannot publish topic: {}" + topic, e);
```
Comment (ctron) [inline kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java]: That should be:

```java
logger.error("Cannot publish topic: {}", topic, e);
```

**Review by ctron — CHANGES_REQUESTED** [review]

**Comment by MMaiero** [comment]
Could you please squash the commits? Thanks

**Comment by dwoodard1** [comment]
Yes, but there's a Squash and Merge option in Github.

**Review by ctron — APPROVED** [review]

**Pull request merged**